src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -393,6 +416,21 @@
       (catch CRLException e
         e))))
 
+(schema/defn set-ca-config-defaults
+  "Sets defaults for any of the missing config parameters for infra CRL"
+  [infraCrlConfig
+   cadir]
+  (log/debug (str (i18n/trs "set-ca-config-defaults received input:")
+                  "\n"
+                  (ks/pprint-to-string infraCrlConfig)))
+  ;; TODO - We want to use the current node/certificate as the default in here instead of starting empty
+  (-> (assoc infraCrlConfig :compile-masters (if (nil? (:compile-masters infraCrlConfig)) [] (:compile-masters infraCrlConfig)))
```

Comment:
I think this would be better with a `merge` rather than `assoc` and a ternary conditional if the value is set.

Reply:
agree current impl is inelegant and verbose. I don't quite understand what you are suggesting and/or how but will give it a try.

Reply:
Not sure if this is what you meant in my latest push.